src/components/ItemsList.js

```diff
@@ -23,13 +32,33 @@ const ItemsList = () => {
     { idField: 'documentId' },
   );
 
-
   const markItemAsPurchased = (index) => {
     const { items, documentId } = shoppingList[0];
-
-    items[index].lastPurchasedOn = items[index].lastPurchasedOn
-      ? null
-      : Date.now();
+    const { lastPurchasedOn } = items[index];
+    const shoppingItemObject = items[index];
+    if (lastPurchasedOn === null) {
+      shoppingItemObject.lastPurchasedOn = Date.now();
+      shoppingItemObject.numberOfPurchases++;
+      shoppingItemObject.daysLeftForNextPurchase = calculateEstimate(
+        undefined,
+        shoppingItemObject.daysLeftForNextPurchase,
+        shoppingItemObject.numberOfPurchases,
+      );
+    } else {
+      if (wasItemPurchasedWithinLastOneDay(lastPurchasedOn)) {
+        return;
+      } else {
+        shoppingItemObject.numberOfPurchases++;
+        const { daysLeftForNextPurchase } = shoppingItemObject;
+        const now = Date.now();
```

Comment:
Again a really small suggestion but since we use the Date.now(); also on line 40, maybe it would be more readable assign the 'now' variable at the top of the function (line 38) and then it can be reused in line 40?

Reply:
Thanks for the suggestion. We can do that for readability though we wanted the recorded time to be as close to the time the data is written to the database as possible. Essentially given the trajectory the app is taking, we shall have to add more lines of code in that function. If we get `Date.now` and take sometime executing many lines of code, our recorded time may not be as precise.

Reply:
We have now created a variable that calls the Date.now() and holds the value for reusability